.devcontainer/Dockerfile

```
@@ -0,0 +1,18 @@
+FROM python:3.12.4-slim
```

Even though we support 3.12, it is recommended to develop on 3.11. 
```suggestion
FROM python:3.11.9-slim
```

There is a few reason for this. 
1. 3.12 allows syntax that is not allowed in earlier versions, i.e., quotes in f-strings.
2. When testing functions, 3.11 is the most recent versions that is allowed.

---

I think this is a very good reason for why a devcontainer is a good idea :) Although you don't need it, it is a very good reference for the environment that PRs should support.